### PR TITLE
Fix!: explode to unnest with multiple explosions

### LINKED ISSUE #2227: semantically incorrect transpilation of unnest from duckdb to bigquery
**Fully reproducible code snippet**
Please include a fully reproducible code snippet or the input sql, dialect, and expected output.

First, the Python:

```
In [19]: import sqlglot as sg

In [20]: sg.parse_one("select unnest(array[1, 2, 3]), unnest(array[4, 5])", read="duckdb").sql(dialect="bigquery")
Out[20]: 'SELECT col, col_2 FROM UNNEST([1, 2, 3]) AS col CROSS JOIN UNNEST([4, 5]) AS col_2'
```

Running this in DuckDB:

```
D select unnest([1, 2, 3]), unnest([4, 5]);
┌──────────────────────────────────┬───────────────────────────────┐
│ unnest(main.list_value(1, 2, 3)) │ unnest(main.list_value(4, 5)) │
│              int32               │             int32             │
├──────────────────────────────────┼───────────────────────────────┤
│                                1 │                             4 │
│                                2 │                             5 │
│                                3 │                               │
└──────────────────────────────────┴───────────────────────────────┘
```

And in the BigQuery UI, because `bq shell` doesn't accept the syntax:

image

This should probably compile to something like this:

```sql
SELECT
  s.col[SAFE_OFFSET(i)] as col,
  s.col_2[SAFE_OFFSET(i)] as col_2
FROM (
  SELECT
    [1, 2, 3] as col,
    [4, 5] as col_2
) as s CROSS JOIN UNNEST(GENERATE_ARRAY(0, GREATEST(ARRAY_LENGTH([1, 2, 3]) - 1, ARRAY_LENGTH([4, 5]) - 1))) AS i
``` 

which gives the same output as DuckDB:

image

**Official Documentation**

https://duckdb.org/docs/sql/query_syntax/unnest.html#unnesting-lists

See the second example

> ```
> -- unnest a scalar list, generating 3 rows ((1, 10), (2, 11), (3, NULL))
> SELECT UNNEST([1, 2, 3]), UNNEST([10, 11]);
> ```

### PR DESCRIPTION
**Pull request by tobymao**
also remove with ordinality with just offset for consistency closes #2227

### CODE REVIEW

File: tests/dialects/test_spark.py [hunk 1]
```diff
@@ -583,3 +583,54 @@ def test_insert_cte(self):
...
+        self.validate_all(
+            "SELECT POSEXPLODE([2, 3])",
+            write={
+                "bigquery": "SELECT IF(pos = pos_2, col, NULL) AS col, IF(pos = pos_2, pos_2, NULL) AS pos_2 FROM UNNEST(GENERATE_ARRAY(0, GREATEST(ARRAY_LENGTH([2, 3])) - 1)) AS pos CROSS JOIN UNNEST([2, 3]) AS col WITH OFFSET AS pos_2 WHERE pos = pos_2 OR (pos > (ARRAY_LENGTH([2, 3]) - 1) AND pos_2 = (ARRAY_LENGTH([2, 3]) - 1))",
+                "presto": "SELECT IF(pos = pos_2, col) AS col, IF(pos = pos_2, pos_2) AS pos_2 FROM UNNEST(SEQUENCE(1, GREATEST(CARDINALITY(ARRAY[2, 3])))) AS _u(pos) CROSS JOIN UNNEST(ARRAY[2, 3]) WITH ORDINALITY AS _u_2(col, pos_2) WHERE pos = pos_2 OR (pos > CARDINALITY(ARRAY[2, 3]) AND pos_2 = CARDINALITY(ARRAY[2, 3]))",
+            },
+        )
+        self.validate_all(
+            "SELECT POSEXPLODE([2, 3]), EXPLODE([4, 5, 6]) FROM tbl",
+            write={
```
Comment (GeorgeSittas) [inline tests/dialects/test_spark.py:626]: This seems like an invalid query in spark v3.4.0:

```
spark-sql (default)> WITH tbl AS (SELECT 1) SELECT POSEXPLODE(ARRAY(2, 3)), EXPLODE(ARRAY(4, 5, 6)) FROM tbl;
[UNSUPPORTED_GENERATOR.MULTI_GENERATOR] The generator is not supported: only one generator allowed per SELECT clause but found 2: "posexplode(array(2, 3))", "explode(array(4, 5, 6))".
```

**Review by GeorgeSittas — COMMENTED** [review]
Looks good at first glance.

Will fix a couple of Spark tests because it needs the `ARRAY(...)` syntax instead of `[...]`, and also left a comment regarding an invalid Spark query.

File: tests/dialects/test_spark.py (hunk 1 above)
Comment (tobymao) [reply tests/dialects/test_spark.py:626]: yes, i know it's invalid

Comment (GeorgeSittas) [reply tests/dialects/test_spark.py:626]: Got it, I left it as is

**Pull request merged**